src/core/Clipboard.cpp

```diff
@@ -22,21 +22,87 @@
  *
  */
 
+#include "Clipboard.h"
+
 #include <QApplication>
 #include <QClipboard>
-#include <QMimeData>
-
-#include "Clipboard.h"
 
+#include "FileBrowser.h"
+#include "PluginFactory.h"
+#include "SampleDecoder.h"
+#include "StringPairDrag.h"
 
 namespace lmms::Clipboard
 {
 
-	const QMimeData * getMimeData()
+	// there are other mimetypes, such as "samplefile", "patchfile" and "vstplugin" but they are generated dynamically.
+	static std::map<std::string, std::vector<std::string>> mimetypes =
 	{
-		return QApplication::clipboard()->mimeData( QClipboard::Clipboard );
+		{"presetfile", {"xpf", "xml", "xiz", "lv2"}},
+		{"midifile", {"mid", "midi", "rmi"}},
+		{"projectfile", {"mmp", "mpt", "mmpz"}},
+	};
+
+	//! gets the extension of a file, or returns the string back if no extension is found
+	inline QString getExtension(const QString& file)
```

Comment:
```suggestion
	static QString getExtension(const QString& file)
```
For internal linkage, since it's not used outside this file